### PR TITLE
Pruner to Kotlin

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java [hunk 1]
```diff
@@ -13,99 +13,89 @@
...
-final class Pruner {
-  final Schema schema;
-  final IdentifierSet identifierSet;
-  final MarkSet marks;
-
+internal class Pruner(
+  private val schema: Schema,
+  private val identifierSet: IdentifierSet
+) {
+  private val marks: MarkSet = MarkSet(identifierSet)
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: supernit: redundant type specification

File: wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java [hunk 2]
```diff
@@ -115,133 +105,139 @@ private void markRoots(ProtoType protoType) {
...
-  private void markReachable() {
-    for (Object root; (root = queue.poll()) != null;) {
-      List<Object> reachableMembers = reachableObjects(root);
-
-      for (Object reachable : reachableMembers) {
-        if (reachable instanceof ProtoType) {
-          if (marks.mark((ProtoType) reachable)) {
-            queue.add(reachable);
+  private fun markReachable() {
+    var root: Any? = queue.poll()
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: call `poll()` only once?

```
while (true) {
  val root = queue.poll() ?: break
  ...
}
```

File: wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java [hunk 3]
```diff
@@ -115,133 +105,139 @@ private void markRoots(ProtoType protoType) {
...
-
-      if (type instanceof MessageType) {
-        Field field = ((MessageType) type).field(member);
-        if (field == null) {
-          field = ((MessageType) type).extensionField(member);
-        }
-        if (field == null) {
-          throw new IllegalStateException("unexpected member: " + member);
+  private fun reachableObjects(root: Any): List<Any?> {
+    val result: MutableList<Any?> = ArrayList()
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: nit: `val result = mutableListOf<Any?>()`

File: wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java [hunk 4]
```diff
@@ -115,133 +105,139 @@ private void markRoots(ProtoType protoType) {
...
+    return result
   }
 
-  private boolean isExtensionField(ProtoMember protoMember) {
-    Type type = schema.getType(protoMember.getType());
-    return type instanceof MessageType
-        && ((MessageType) type).extensionField(protoMember.getMember()) != null;
+  private fun isExtensionField(protoMember: ProtoMember): Boolean {
+    val type = schema.getType(protoMember.type)
+    return (type is MessageType && type.extensionField(protoMember.member) != null)
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: nit: extra parens

File: wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java (hunk 2 above)
Comment (oldergod) [reply wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: of course!

**Comment by oldergod** [comment]
good to merge on green

**Pull request merged**